### PR TITLE
Don't add a download multiple times

### PR DESCRIPTION
**Pull request by TobiasKiehnlein**
Prevent users to add the same download multiple times which results in unwanted behavior.

Adjust minor code inconsistencies and remove duplicated code.

### CODE REVIEW

File: diggicamp/application/basics.py [hunk 1]
```diff
@@ -57,30 +59,10 @@ def add_download(dgc: Diggicamp, folder_id: str, target: str, regex=None):
 
     if not dgc.conf.get('downloads'):
         dgc.conf.set('downloads', [entry])
-    else:
+    elif not any(map(lambda download_entry: download_entry[download_type] == download_id, dgc.conf.get('downloads'))):
```
Comment (AntonLydike) [inline diggicamp/application/basics.py:62]: This is a little bit clunky to read, I would suggest:
``` python
not any((entry[download_type] == download_id for entry in dgc.conf.get('downloads')))
```

This just passes an iterator and expresses the `map` as an iterator comprehension. It should be about as fast as the list is still lazily evaluated. 

Alternatively, this approach might be even more readable (but is definitely slower for large inputs):

```python3
download_id not in [entry[download_type] for entry in dgc.conf.get('downloads')]
```

All these are untested, but I'm pretty sure they work.

**Review by AntonLydike — CHANGES_REQUESTED** [review]
The rest looks good, except for a few minor issues. I'm currently still recovering from covid, but I'll try to test this as soon as possible!

File: .gitignore [hunk 2]
```diff
@@ -2,3 +2,4 @@ __pycache__
 *.pyc
 dgc.json
 venv
+.idea
```
[the remark this answers is not in the document]
Comment (TobiasKiehnlein) [reply .gitignore]: Updated the `.gitignore` using the recommendations that can be found on [github](https://github.com/github/gitignore/blob/main/Global/JetBrains.gitignore).

This results in some configuration files for PyCharm which have now been pushed for the sake of consistency.

File: diggicamp/application/basics.py (hunk 1 above)
Comment (TobiasKiehnlein) [reply diggicamp/application/basics.py]: Committed the first proposal due to the best compromise between readability and performance.

File: .gitignore (hunk 2 above)
[the remark this answers is not in the document]
Comment (AntonLydike) [reply .gitignore]: Yes, almost! The Jetbrains tools actually provide their own `.gitignore` located at `.idea/.gitignore` which gets updated when Jetbrains updates their internal structure. This is slightly preferable as it is less likely to become out-of-date. This means you can remove all jetbrains-specific stuff from your `.gitignore` files. 

During the initial review, I didn't notice that this repository wasn't even a proper Jetbrains Project, my bad.

File: diggicamp/application/basics.py (hunk 1 above)
Comment (AntonLydike) [reply diggicamp/application/basics.py]: Awesome! Thanks!

**Review by AntonLydike — APPROVED** [review]
Looks pretty good!

**Comment by AntonLydike** [comment]
Everything looks good! Thanks for the contribution!

**Pull request merged**